noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/components/public_tail_output_validator.nr

```diff
@@ -214,10 +214,35 @@ impl<let NUM_PUBLIC_DATA_LEAVES: u32> PublicTailOutputValidator<NUM_PUBLIC_DATA_
         );
 
         // Check that the output contains the last write in each linked list.
-        assert_deduped_array(
-            self.hints.public_data_writes,
-            self.output.end.public_data_update_requests,
-        );
+        let original_array = self.hints.public_data_writes;
```

Comment:
Copied the code from `assert_deduped_array`, and tweaked it so that it operates on the deduped array that contains different types to the items in the original array.

Can't be bothered to change `assert_deduped_array` since we will remove the public kernel soon.